src/emm/services/github_service.py

```diff
@@ -0,0 +1,61 @@
+"""Service for working with github CLI."""
+from pathlib import Path
+from typing import List, Optional
+
+import plumbum
+from plumbum import local
+
+
+class GithubService:
```

Comment:
Why doesn't this service follow the same pattern as the rest of the cli services? In all the other services, we initialize an object with the `local.cmd.<command name>` that is then used in all the methods. This makes things much easier to test as we don't have to patch `local.cmd.<command name>` in all the tests.

Reply:
I have used that before, but this method requires the testing machine in evergreen install the github CLI. Since we have 3 different OS to test, and we are expecting users to install the github CLI by themselves. 
I think this is a more safer and easier way for us to test locally and in evergreen. Otherwise, it would require a lot more dependencies such homebrew in mac OS and handle strange case in windows.

Reply:
We should still mock out the binary so it isn't needed on the system running the tests, but my point is that we should inject that mock in the constructor and use that instead of every test needing to patch the mock. I generally consider using `@patch` to be a bad practice. There are times when it is needed, but we should avoid it if we can. We have been using [Dependency Inject](https://en.wikipedia.org/wiki/Dependency_injection) to try and avoid needing to patch, which tends to result in cleaner code that is easier to test.